Server/MirEnvir/Envir.cs

```diff
@@ -2532,7 +2533,93 @@ public bool CharacterExists(string name)
 
             return false;
         }
+        public List<CharacterInfo> MatchPlayer(string PlayerID, bool match = false)
+        {
+            if (string.IsNullOrEmpty(PlayerID)) return new List<CharacterInfo>(CharacterList);
+
+            List<CharacterInfo> list = new List<CharacterInfo>();
+
+            for (int i = 0; i < CharacterList.Count; i++)
+            {
+                if (match)
+                {
+                    if (CharacterList[i].Name.Equals(PlayerID, StringComparison.OrdinalIgnoreCase))
+                        list.Add(CharacterList[i]);
+                }
+                else
+                {
+                    if (CharacterList[i].Name.IndexOf(PlayerID, StringComparison.OrdinalIgnoreCase) >= 0)
+                        list.Add(CharacterList[i]);
+                }
+            }
+
+            return list;
+        }
+        public List<CharacterInfo> MatchPlayerbyItem(string ItemID, bool match = false)
+        {
+            if (string.IsNullOrEmpty(ItemID)) return new List<CharacterInfo>(CharacterList);
+
+            ulong itemId = 0;
+            ulong.TryParse(ItemID, out itemId);
```

Comment:
you should probably just return nothing if the id isn't a number

Reply:
I've just removed the null check. If a letter is input then it returns nothing and if a unused item id is input it does the same.

Reply:
yes but why bother doing the rest of the code when you know 0 will never find anything?

if its not parsed as a number, or if 0 is entered - it should just return no results straight away

Reply:
I see, I'll do a pr

Reply:
![image](https://github.com/Suprcode/Crystal/assets/68875342/d0501d9f-3027-4d46-be98-44f20a64b171)

Reply:
Move the list above it and just return the empty list